ssdp.md

```diff
@@ -0,0 +1,135 @@
+1. when a service is no more available, the root device needs to advertise a "NOTIFY" message from type "ssdp:byebye" with information about the corresponding service. Control points can remove the service if it is listed. 
+
+# Evaluation
+
+this section evaluates SSDP as discovery protocol for the OpenScreenProtocol according several functional and non-functional requirements
+
+## Functional Requirements
+
+We will consider the functional requirements of the Presentation API (and later also Remote Playback API). The functional requirement related to discovery is the ability to "Monitor Display Availability" which is described in section [6.4 Interface PresentationAvailability](https://w3c.github.io/presentation-api/#interface-presentationavailability) of the Presentation API specification. 
+
+### Presentation API: Monitor Display Availability
+
+The entry point in the Presentation API to monitor display availability is the [PresentationRequest](https://w3c.github.io/presentation-api/#interface-presentationrequest) interface. The algorithm [Monitoring the list of available presentation displays](https://w3c.github.io/presentation-api/#dfn-monitor-the-list-of-available-presentation-displays) is used in [PresentationRequest.start()](https://w3c.github.io/presentation-api/#dom-presentationrequest-start) an in [PresentationRequest.getAvailability()] (https://w3c.github.io/presentation-api/#dom-presentationrequest-getavailability) methods. Only presentation displays that can open at least one of the URLs passed as input in the PresentationRequest constructor should be considered. There are two methods how SSDP can be used to monitor display availability in Presentation API:
+
+**Method 1**: It is similar to of SSDP as discovery in DIAL. The main steps are listed below:
+
+* The display advertises the presentation receiver service when it is added on the network using SSDP and the service type `urn:example-org:service:osp:1` (this is just an example). The SSDP alive message contains a LOCATION header which points to the XML device description (contains friendly name, device capabilities, etc.). The display should advertise a SSDP byebye message before it is not more available. 
+* The controller starts "Monitor Display Availability" algorithm by sending a SSDP search message with the service type `urn:example-org:service:osp:1` and waits for responses from devices running a presentation receiver service. The controller should wait for SSDP alive/byebye messages on the multicast address to keep the list of available displays up-to-date (e.g. when a new display in added or an existing one is removed).  
+* Each display available on the network and runs a presentation receiver service replies to the search request with a SSDP message similar to the alive message.
+* The controller reads the LOCATION header for each available display and makes a HTTP GET request to get the device description XML. 
```

Comment:
Adding a dependency on HTTP+XML adds some security risks and complexity from an implementer perspective.

- Untrusted XML parsing is generally unsafe and should be done in a sandboxed process.
- The HTTP requests have to be done carefully since they bypass all of the normal origin related checks that you would get from Web content.